Automattic/jetpack
# build: Only build changed projects (and their dependents/dependencies)

#### Changes proposed in this Pull Request:

If we have a PR touching Boost, for example, there's no point in
building Jetpack too.

#### Jetpack product discussion


PT: p9dueE-4dJ-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* See that the build on this PR builds everything.
* Create a PR based on this one that only touches one thing. See that only that thing and its dependents and dependencies are built.
  * :heavy_check_mark: https://github.com/Automattic/jetpack/actions/runs/1808943454 skipped the github-actions, js-packages/eslint-changed, and most of the plugins for example.
  * :heavy_check_mark: https://github.com/Automattic/jetpack/actions/runs/1808989344 only built packages/changelogger, packages/autoloader, and plugins/beta
* Create a PR based on this one that only touches non-projects (and not any of the infrastructure files either). See that nothing is built, and that nothing breaks because of that.
  * :heavy_check_mark: https://github.com/Automattic/jetpack/actions/runs/1809001177
Closes #22721
